### PR TITLE
docs: update android setup with react native 0.71

### PR DESCRIPTION
**Pull request by xiongemi**
## Description

- This pull request addresses the issue described here: #<?>

In this pull request, I have updates the docs for android setup with react native 71

---

> _For features/enhancements:_
 - [x] I have added/updated the relevant references in the [documentation](https://github.com/wix/Detox/tree/master/docs) files.

> _For API changes:_
 - [x] I have made the necessary changes in the [types index](https://github.com/wix/Detox/blob/master/detox/index.d.ts) file.

### CODE REVIEW

File: docs/introduction/project-setup.mdx [hunk 1]
```diff
@@ -150,36 +150,35 @@ patch or create if they are missing:
...
      minSdkVersion = 21 // (1)
-     compileSdkVersion = 30
-     targetSdkVersion = 30
+     compileSdkVersion = 33
+     targetSdkVersion = 33
 // highlight-next-line
-+    kotlinVersion = 'X.Y.Z' // (2)
++    kotlinVersion = "X.Y.Z" // (2)
+     // We use NDK 23 which has both M1 support and is the side-by-side NDK version from AGP.
+      ndkVersion = "23.1.7779620"
```
Comment (noomorph) [inline docs/introduction/project-setup.mdx:163]: Isn't the version too specific? I guess just after 2-4 weeks the build number `7779620` will be different when someone installs the latest 23.x NDK. @d4vidi , @xiongemi please advise.

File: docs/introduction/project-setup.mdx [hunk 2]
```diff
@@ -150,36 +150,35 @@ patch or create if they are missing:
...
-     classpath("de.undercouch:gradle-download-task:5.0.1")
 // highlight-next-line
 +    classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion") // (3)
  …
-
- allprojects {
-   repositories {
-   …
-   google()
+}
```
Comment (noomorph) [inline docs/introduction/project-setup.mdx:172]:
```suggestion
}

```

Maybe better to separate with an extra newline.

File: docs/introduction/project-setup.mdx [hunk 3]
```diff
@@ -150,36 +150,35 @@ patch or create if they are missing:
...
+allprojects {
+    repositories {
```
Comment (noomorph) [inline docs/introduction/project-setup.mdx:174]: Could you please keep indent level at 2?

File: docs/introduction/project-setup.mdx [hunk 4]
```diff
@@ -150,36 +150,35 @@ patch or create if they are missing:
...
 // highlight-start
-+  maven { // (4)
-+    url("$rootDir/../node_modules/detox/Detox-android")
-+  }
++       maven { // (4)
++           // https://wix.github.io/Detox/docs/introduction/project-setup
```
Comment (noomorph) [inline docs/introduction/project-setup.mdx:177]:
```suggestion
```

I guess this comment is redundant here.

**Review by noomorph — CHANGES_REQUESTED** [review]

**Comment by d4vidi** [comment]
We have known issues with SDK v33 (so can't recommend it in the docs) and are not officially compliant with RN v0.71 yet.
I have to close this in favor of #3884, which we will get to sometime in Q2.

**Pull request closed without merging**